bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/discovery/SunspecDiscoveryProcess.java

```diff
@@ -0,0 +1,425 @@
+
+/**
+ * This class is used by the SunspecDiscoveryParticipant to detect
+ * the model blocks defined by the given device.
+ * It scans trough the defined model items and notifies the
+ * discovery service about the discovered devices
+ *
+ * @author Nagy Attila Gabor - Initial contribution
+ */
+@NonNullByDefault
+public class SunspecDiscoveryProcess {
+
+    /**
+     * Logger instance
+     */
+    private final Logger logger = LoggerFactory.getLogger(SunspecDiscoveryProcess.class);
+
+    /**
+     * The handler instance for this device
+     */
+    private final ModbusEndpointThingHandler handler;
+
+    /**
+     * The endpoint where we can reach the device
+     */
+    private final Optional<ModbusSlaveEndpoint> endpoint;
+
+    /**
+     * Listener for the discovered devices. We get this
+     * from the main discovery service, and it is used to
+     * submit any discovered Sunspec devices
+     */
+    private final ModbusDiscoveryListener listener;
+
+    /**
+     * The endpoint's slave id
+     */
+    private Integer slaveId;
+
+    /**
+     * Number of maximum retries
+     */
+    private Integer maxTries = 3;
+
+    /**
+     * List of start addresses to try
+     */
+    private List<Integer> possibleAddresses;
+
+    /**
+     * This is the base address where the next block should be searched for
+     */
+    private int baseAddress = 40000;
+
+    /**
+     * Count of valid Sunspec blocks found
+     */
+    private int blocksFound = 0;
+
+    /**
+     * Parser for commonblock
+     */
+    private final CommonModelParser commonBlockParser;
+
+    /**
+     * The last common block found. This is used
+     * to get the details of any found devices
+     */
+    private Optional<CommonModelBlock> lastCommonBlock = Optional.empty();
+
+    /**
+     * New instances of this class should get a reference to the handler
+     *
+     * @throws EndpointNotInitializedException
+     */
+    public SunspecDiscoveryProcess(ModbusEndpointThingHandler handler, ModbusDiscoveryListener listener)
+            throws EndpointNotInitializedException {
+        this.handler = handler;
+        ModbusSlaveEndpoint endpoint = this.handler.asSlaveEndpoint();
+        if (endpoint != null) {
+            this.endpoint = Optional.of(endpoint);
+        } else {
+            this.endpoint = Optional.empty();
+        }
+        slaveId = handler.getSlaveId();
+        this.listener = listener;
+        commonBlockParser = new CommonModelParser();
+        possibleAddresses = new CopyOnWriteArrayList<>();
+        // Preferred and alternate base registers
+        // @see SunSpec Information Model Overview
+        possibleAddresses.add(40000);
+        possibleAddresses.add(50000);
+        possibleAddresses.add(0);
+    }
+
+    /**
+     * Set the maximum number of retries for operations
+     *
+     * @param num the new value to set
+     * @return
+     */
+    public SunspecDiscoveryProcess setMaxTries(Integer num) {
```

Comment:
Is this used?

Reply:
not at the moment. Removed, I can add it back later if there will be a need to set this value.
According to this changed the maxTries property to static final